docs/userguides/deactivatedevices.md

```diff
@@ -38,6 +38,9 @@ curating a list of devices to deactivate:
 * `--exclude-most-recently-connected INTEGER` - allows you to exclude the most recently connected device (per user) from the results. This allows you to ensure that every user is left with at least N device(s), regardless of how recently they have connected.
 * `--created-before DATE|TIMESTAMP|SHORT_TIME` - allows you to only see devices created before a particular date.
 
+```eval_rst
```

Comment:
I noticed this also happens with `userUid`s... I am not sure if this is the best place for this note.

Reply:
Perhaps it'd be a good idea to have a whole guide dedicated to working with CSVs.  It can contain examples for outputting CSV files and using CSV files in bulk commands. It can mention potential issues and workarounds, such as dealing with scientific notation false auto-detection.

Reply:
could this be considered a bug? would we see this behavior if all these types of ids got exported to the csvs wrapped in quotes?

Reply:
@unparalleled-js to more directly answer your question, I think that if there are issues related to csvs in general rather than specific cli commands that it's worth having their own section. Otherwise we'll find ourself repeating this info for user uids, device uids, etc, etc. If the problem is the format and not the command, let's have the guidance be reflective of that.

It still may be helpful to have a note like:
```
This command exports data to a csv file. To troubleshoot common issues when exporting to csv, see `here`
```

Near any potentially impacted commands.

Reply:
I just tested with numbers wrapped in quotes and Excel still rounded them, so I don't think that would make a difference.

Reply:
I wonder if it'd be possible to detect that the IDs might have been modified. Like if when we hit a "not found" error for any "long" ID type, we could check to see if the last digits are all `0`s and log a message indicating Excel as possibly the culprit?

Reply:
> could this be considered a bug? would we see this behavior if all these types of ids got exported to the csvs wrapped in quotes or something? I'd be curious what the difference is between this file as-is and when it is opened into a new playbook.

I doubt this would be considered a bug. I don't think that *we* are doing anything unusual, it's more of just the result of how Excel chooses to interpret those CSVs. Searching for "long number csv excel" shows a lot of other applications having similar issues with Excel, and the standard fix/workaround has basically always been "do an import, don't open directly." While Excel is almost certainly the most widely used software for our customers to view/manipulate those CSVs, it's not the only tool (I use the [Visidata](https://www.visidata.org/) utility mostly, myself) and doing things "the Excel way" may result in other tools being confused or having to remove excess characters, like quotes or backticks.

I don't actually *have* an Excel license for testing, but it looks like one may be able to coerce Excel into not changing the format on open by [various kludgy tricks](https://superuser.com/questions/861964/how-do-i-stop-excel-from-converting-value-0503e000-to-5-03e02-automatically/862073) (e.g. `="136479852364713598"` or ``` `13518797813518```)


> It still may be helpful to have a note like:
> 
> ```
> This command exports data to a csv file. To troubleshoot common issues when exporting to csv, see `here`
> ```
> 
> Near any potentially impacted commands.

Definitely. A "Working with CSVs" seems like a useful thing to have, but we'd definitely want to have a link to that from any articles directly. I don't think that people would likely think of the CSV being the culprit when something went wrong, and think to look for that.

Reply:
I didn't realize this but we have a [Import CSV Reports to Microsoft Excel](https://support.code42.com/Administrator/Cloud/Monitoring_and_managing/Import_CSV_reports_to_Microsoft_Excel) guide in the official Code42 documentation...perhaps this should be rewritten to point to that guide instead.